src/analyses/variable-sensitivity/variable_sensitivity_domain.cpp

```diff
@@ -0,0 +1,345 @@
+void variable_sensitivity_domaint::output(
+  std::ostream &out,
+  const ai_baset &ai,
+  const namespacet &ns) const
+{
+  abstract_state.output(out, ai, ns);
+}
+
+/*******************************************************************\
+
+Function: variable_sensitivity_domaint::make_bottom
+
+  Inputs: None
+
+ Outputs: None
+
+ Purpose: Sets the domain to bottom (no relations).
+
+\*******************************************************************/
+void variable_sensitivity_domaint::make_bottom()
+{
+  abstract_state.make_bottom();
+  return;
+}
+
+/*******************************************************************\
+
+Function: variable_sensitivity_domaint::make_top
+
+  Inputs: None
+
+ Outputs: None
+
+ Purpose: Sets the domain to top (all relations).
+
+\*******************************************************************/
+void variable_sensitivity_domaint::make_top()
+{
+  abstract_state.make_top();
+}
+
+
+/*******************************************************************\
+
+Function: variable_sensitivity_domaint::make_entry
+
+  Inputs: None
+
+ Outputs: None
+
+ Purpose: Set up a sane entry state.
+
+\*******************************************************************/
+void variable_sensitivity_domaint::make_entry()
+{
+  abstract_state.make_top();
+}
+
+/*******************************************************************\
+
+Function: variable_sensitivity_domaint::merge
+
+  Inputs: The other domain (b) and it's preceding location (from) and
+          current location (to).
+
+ Outputs: True if something has changed.
+
+ Purpose: Computes the join between "this" and "b".
+
+\*******************************************************************/
+
+bool variable_sensitivity_domaint::merge(
+  const variable_sensitivity_domaint &b,
+  locationt from,
+  locationt to)
+{
+  #ifdef DEBUG
+  std::cout << "merging " << from->source_location.get_line() << "->"
+            << to->source_location.get_line() << std::endl;
+  #endif
+
+  // Use the abstract_environment merge
+  bool any_changes=abstract_state.merge(b.abstract_state);
+
+  assert(abstract_state.verify());
+  return any_changes;
+}
+
+/*******************************************************************\
+
+Function: variable_sensitivity_domaint::ai_simplify
+
+  Inputs:
+   condition - the expression to simplify
+   ns - the namespace
+   lhs - is the expression on the left hand side
+
+ Outputs: True if simplified the condition. False otherwise. condition
+          will be updated with the simplified condition if it has worked
+
+ Purpose: Use the information in the domain to simplify the expression
+          with respect to the current location.  This may be able to
+          reduce some values to constants.
+
+\*******************************************************************/
+
+bool variable_sensitivity_domaint::ai_simplify(
+  exprt &condition, const namespacet &ns, const bool lhs) const
+{
+  if (lhs)
```

Comment:
To clarify the scope of this should be moved into a separate function `ai_simplify_lhs`